tests/beaker_tests/cisco_portchannel_global/test_portchannel_global.rb

```diff
@@ -12,509 +12,323 @@
+    hash_poly:         'CRC10c',
+    load_defer:        '1000',
+    resilient:         'true',
+    rotate:            '4',
+    symmetry:          'true',
   },
 }
 
-tests['non_default_properties_no_hash'] = {
-  title_pattern:  'default',
-  manifest_props: "
-    bundle_hash                  => 'ip',
-    bundle_select                => 'dst',
-    rotate                       => '4',
-  ",
-  resource_props: {
-    'bundle_hash'   => 'ip',
-    'bundle_select' => 'dst',
-    'rotate'        => '4',
+# Per platform non default values
+manifest_non = {
+  n3k:  {
+    bundle_hash:   'ip-only',
+    bundle_select: 'src-dst',
+    resilient:     'false',
   },
-}
-
-tests['default_properties_no_rotate'] = {
-  title_pattern:  'default',
-  manifest_props: "
-    bundle_hash                  => 'default',
-    bundle_select                => 'default',
-    resilient                    => 'default',
-    symmetry                     => 'default',
-  ",
-  code:           [0, 2],
-  resource_props: {
-    'bundle_hash'   => 'ip',
-    'bundle_select' => 'src-dst',
-    'resilient'     => 'true',
-    'symmetry'      => 'false',
+  n56k: {
+    bundle_hash: 'mac'
   },
-}
-
-tests['non_default_properties_no_rotate'] = {
-  title_pattern:  'default',
-  manifest_props: "
-    bundle_hash                  => 'ip-only',
-    bundle_select                => 'src-dst',
-    resilient                    => 'false',
-    symmetry                     => 'true',
-  ",
-  resource_props: {
-    'bundle_hash'   => 'ip-only',
-    'bundle_select' => 'src-dst',
-    'resilient'     => 'false',
-    'symmetry'      => 'true',
+  n8k:  {
+    bundle_hash: 'ip'
+  },
+  n9k:  {
+    bundle_hash:   'ip',
+    bundle_select: 'src-dst',
   },
 }
 
-#################################################################
-# HELPER FUNCTIONS
-#################################################################
-
-# Full command string for puppet resource command
-def puppet_resource_cmd
-  PUPPET_BINPATH + 'resource cisco_portchannel_global'
-end
-
-def build_manifest_portchannel_global(tests, id)
-  manifest = tests[id][:manifest_props]
-  tests[id][:resource] = tests[id][:resource_props]
-
-  tests[id][:title_pattern] = id if tests[id][:title_pattern].nil?
-  logger.debug(
-    "build_manifest_portchannel_global :: title_pattern:\n" +
-             tests[id][:title_pattern])
-  tests[id][:manifest] = "cat <<EOF >#{PUPPETMASTER_MANIFESTPATH}
-  node 'default' {
-    cisco_portchannel_global { 'default':
-      #{manifest}
-    }
-  }
-EOF"
+tests[:non_default][:manifest_props].merge!(manifest_non[:n3k]) if platform[/n3k/]
+tests[:non_default][:manifest_props].merge!(manifest_non[:n56k]) if platform[/n(5|6)k/]
+tests[:non_default][:manifest_props].merge!(manifest_non[:n8k]) if platform[/n8k/]
+tests[:non_default][:manifest_props].merge!(manifest_non[:n9k]) if platform[/n9k/]
+
+def unsupported_properties(tests, _id)
+  unprops = []
+  if platform[/n7k/]
```

Comment:
nit: I prefer a case here instead of all these elsif's. I won't make you change it though :-)